index.d.ts

```diff
@@ -0,0 +1,41 @@
+declare module 'react-swipeable' {
+  import * as React from 'react'
+
+  namespace ReactSwipeable {
+    interface Event<T> {
+      event: React.TouchEvent<T>
+      deltaX: number
+      deltaY: number
+      absX: number
+      absY: number
+      velocity: number
+      dir: 'Left' | 'Right' | 'Up' | 'Down'
+    }
+
+    type Callback<T extends Element = HTMLElement> = ({}: Event<T>) => void
+
+    interface SwipeableProps<T extends Element = HTMLElement> extends React.ClassAttributes<Swipeable<T>>, React.HTMLAttributes<T> {
```

Comment:
the Swipeable Props again are explicit and aren't extending anything, i think this too may be a bit overkill and could just be a simple object interface?

Reply:
Yep, carried the generics over from the [DefinitelyTyped definitions](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/react-swipeable/index.d.ts). I've remove all the generics I can without my editor complaining